include/xgboost/c_api.h

```diff
@@ -1386,4 +1388,86 @@ XGB_DLL int XGBoosterFeatureScore(BoosterHandle handle, const char *json_config,
                                   bst_ulong *out_dim,
                                   bst_ulong const **out_shape,
                                   float const **out_scores);
+
+/*!
+ * \brief initialize the collective communicator,
+ *  call this once before using anything
+ *  The additional arguments is not necessary.
+ *  Usually the communicator will detect settings
+ *  from environment variables.
+ * \param argc number of arguments in argv
+ * \param argv the array of input arguments
+ */
+XGB_DLL int XGCommunicatorInit(int argc, char *argv[]);
```

Comment:
We have been using JSON for this type of argument. I think it's better than argv since the input can be easily verified.

Reply:
Done.